renderdoc/os/posix/linux/linux_process.cpp

```diff
@@ -35,6 +38,42 @@ char **GetCurrentEnvironment()
   return environ;
 }
 
+std::vector<int> getSockets(pid_t childPid)
+{
+  std::vector<int> sockets;
+  string dirPath = StringFormat::Fmt("/proc/%d/fd", (int)childPid);
+  DIR *pDir = opendir(dirPath.c_str());
```

Comment:
Rather than reproducing `opendir` / `readdir` iteration here I'd prefer to use the existing `FileIO::GetFilesInDirectory`. From a quick test this should still provide you the information you want and you can then do `readlink` to see which fd is a socket.